packages/scope-manager/tests/fixtures/type-declaration/mapped-named.ts.shot

```diff
@@ -41,7 +41,6 @@ ScopeManager {
       defs: [
         TypeDefinition$3 {
           name: Identifier<"k">,
-          node: TSTypeParameter$3,
```

Comment:
...except, I suspect the removal of the type parameter node from the type definition is probably wrong?

Reply:
yeah this should still be there - everything must have a `node`.
This is because of your code here:
https://github.com/JoshuaKGoldberg/typescript-eslint/blob/210e2810f6e83be0f3b565cb16bf476a05d945dd/packages/scope-manager/src/referencer/TypeVisitor.ts#L208-L211

In the scope analyser we don't actually have access to the `.parent` cos that property is actually set by ESLint during traversal! The AST we're analysing only goes down.

So `node.parent` is `undefined` - hence this snapshot serialises without its `node`.

Reply:
👍 makes sense. And now it's back to existing, this time as the `TSMappedType$*` itself. Great!